src/internal.c

```diff
@@ -20703,12 +21282,32 @@ static int SendAlert_ex(WOLFSSL* ssl, int severity, int type)
      * TLS 1.3 encrypts handshake packets after the ServerHello
      */
     if (IsEncryptionOn(ssl, 1)) {
+#ifdef WOLFSSL_DTLS13
+        if (ssl->options.dtls
+            && IsAtLeastTLSv1_3(ssl->version)
+            && !w64Equal(ssl->dtls13EncryptEpoch->epochNumber, ssl->dtls13Epoch)) {
+            ret = Dtls13SetEpochKeys(ssl, ssl->dtls13Epoch, ENCRYPT_SIDE_ONLY);
+            if (ret != 0)
+                return ret;
+        }
+#endif /* WOLFSSL_DTLS13 */
+
         sendSz = BuildMessage(ssl, output, outputSz, input, ALERT_SIZE, alert,
                                                                        0, 0, 0, CUR_ORDER);
     }
     else {
 
-        AddRecordHeader(output, ALERT_SIZE, alert, ssl, CUR_ORDER);
+        if (ssl->options.dtls && IsAtLeastTLSv1_3(ssl->version)) {
+#ifdef WOLFSSL_DTLS13
+            ret = Dtls13RlAddPlaintextHeader(ssl, output, alert, ALERT_SIZE);
+            if (ret != 0)
+                return ret;
+#endif /* WOLFSSL_DTLS13 */
```

Comment:
Should there be an `#else` case here? Otherwise this just skips.

Reply:
If we are in the `if` branch, then dtlsv1.3 should be supported, otherwise we should get in the `else` branch. 
Guarding the body of the `if` looks nicer to me but I agree with you that can be dangerous or misleading. I changed it to follow:
```
#ifdef guard
if (condition) {
}
else
#endif 
{
}
```
usual pattern.